### PR TITLE
Checkout PR head for strict error checker

### PR DESCRIPTION
**Pull request by Johennes**
On https://github.com/matrix-org/matrix-react-sdk/pull/10083 I currently have the issue that the typescript strict check fails complaining about violations in files that I didn't touch in the PR. I _think_ this is because by default the `checkout` action checks out the repository's default branch and then merges the PR branch into it before running the checks. This might be desirable in other contexts but I think for the strict mode checks we really just care about issues in files that the PR actually touched.

---
This change is marked as an *internal change* (Task), so will not be included in the changelog.

### CODE REVIEW

**Comment by Johennes** [comment]
Hm, I guess I'm not sure why the jobs are failing here. 🤔 Will need to look into this later. Making this a draft for now.

**Comment by t3chguy** [comment]
@Johennes because you didn't run prettier.

```
[warn] .github/workflows/static_analysis.yaml
[warn] Code style issues found in the above file. Forgot to run Prettier?

File: .github/workflows/static_analysis.yaml [hunk 1]
```diff
@@ -52,6 +52,8 @@ jobs:
                     - "--noImplicitAny"
         steps:
             - uses: actions/checkout@v3
+              with:
+                ref: ${{ github.event.pull_request.head.sha }}
```
Comment (t3chguy) [inline .github/workflows/static_analysis.yaml:56]:
```suggestion
                  ref: ${{ github.event.pull_request.head.sha }}
```

Comment (Johennes) [reply .github/workflows/static_analysis.yaml:56]: Oh, well this is what happens when committing from the web UI in-between two other tasks. 🤦

Thanks.

**Pull request merged**